tests/test_cross_validation.cc

```diff
@@ -19,7 +19,47 @@
 
 namespace albatross {
 
-TYPED_TEST_P(RegressionModelTester, test_predict_variants) {
+// Group values by interval, but return keys that once sorted won't be
+// in order
+std::string group_by_interval(const double &x) {
+  if (x <= 3) {
+    return "2";
+  } else if (x <= 6) {
+    return "3";
+  } else {
+    return "1";
+  }
+}
+
+// Group values by interval, but return keys that once sorted won't be
+// in order
+std::string group_by_interval(const AdaptedFeature &x) {
+  return group_by_interval(x.value);
+}
+
+bool is_monotonic_increasing(const Eigen::VectorXd &x) {
+  for (Eigen::Index i = 0; i < x.size() - 1; i++) {
+    if (x[i + 1] - x[i] <= 0.) {
+      return false;
+    }
+  }
+  return true;
+}
+
+TYPED_TEST_P(RegressionModelTester, test_logo_predict_variants) {
+  auto dataset = this->test_case.get_dataset();
+  auto model = this->test_case.get_model();
+
+  LeaveOneGroupOut<typename decltype(dataset)::Feature> logo(group_by_interval);
+  const auto prediction =
+      model.cross_validate().predict(dataset, logo);
+
+  EXPECT_TRUE(is_monotonic_increasing(prediction.mean()));
```

Comment:
It's hard to tell if this test is valid or not without seeing the construction to `test_case.`  Is the idea that cross validated predictions should be monotonically increasing, independent of the groupFunction? So one purposefully designed to rearrange the features checks this best?

Reply:
Yeah, the idea is that the dataset is linear but the group function will not preserve order so this is testing to make sure cross validation deals with unsorted groups appropriately.  I added a comment to clarify.